Migration refactor

Minny27 opened the pull request:
## Motivation ⍰

- 스플래시 슈퍼 뷰 백그라운드 색상 설정
- 탭 바 높이, 아이템 커스텀

<br>

## Key Changes 🔑

- 

<br>

## To Reviewers 🙏🏻

- [ ] 스플래시 화면 이전이랑 동일하게 나오는 지 확인해주세요
- [ ] 노치 유무에 따라 탭 바 높이가 일정한 지, 탭 바 아이템이 정상적으로 나오는 지 확인해주세요

<br>

## Linked Issue 🔗

-

Minny27 committed:
Refactor: 스플래시 뷰의 슈퍼 뷰 배경 색 설정

Minny27 committed:
Refactor: 이미지의 파일 네임 스페이스를 가지도록 수정

Minny27 committed:
Refactor: 개행 수정

Minny27 committed:
Enhancement: 탭 바 수정

- 높이 56 기본 크기 및 노치에 따른 높이 설정
- 바 아이템 텍스트 색상 설정
- 각각의 VC에서 tab bar item 설정

Minny27 commented on Projects/Features/Like/Src/Home/LikeHomeViewController.swift:
```diff
@@ -40,14 +40,22 @@ public final class LikeHomeViewController: TFBaseViewController {
+    tabBarItem = UITabBarItem(
+      title: "하트",
+      image: DSKitAsset.Image.Tab.heart.image.withRenderingMode(.alwaysOriginal),
+      selectedImage: DSKitAsset.Image.Tab.heartSelected.image.withRenderingMode(.alwaysOriginal))
```
Tab Bar 클래스에서 case 나누어서 Tab Bar Item을 설정할까 하다가
네비 바 아이템 설정과 비슷하게 각 Feature에서 설정하는 방식으로 수정했습니다.

Minny27 commented on Projects/Features/Like/Src/Home/LikeHomeViewController.swift:
```diff
@@ -40,14 +40,22 @@ public final class LikeHomeViewController: TFBaseViewController {
 
     navigationItem.title = "나를 좋아요한 무디"
-    let noti = UIBarButtonItem(image: DSKitAsset.Image.bell.image, style: .plain, target: nil, action: nil)
+    let noti = UIBarButtonItem(image: DSKitAsset.Image.Icons.bell.image, style: .plain, target: nil, action: nil)
```
이미지 에셋 안에 폴더링이 되어 있는데, 폴더가 인식되도록 nameSpace 포함하도록 수정했습니다.

Minny27 commented on Projects/Features/Src/Main/TFTabBarViewController.swift:
```diff
@@ -27,28 +28,36 @@ final class TFTabBarController: UITabBarController, MainViewControllable {
+  
   func setViewController(_ viewControllables: [ViewControllable]) {
     super.setViewControllers(viewControllables.map { $0.uiController }, animated: false)
+    setAppearance()
```
이전 main과 동일하게 탭 바의 Appearance 설정과 탭 바 아이템 텍스트 설정 코드 추가했습니다.

Minny27 commented on Projects/Features/Src/Main/TFTabBarViewController.swift:
```diff
@@ -13,6 +13,7 @@ import DSKit
 
 final class TFTabBarController: UITabBarController, MainViewControllable {
   var uiController: UIViewController { self }
+  let tabBarHeight: CGFloat = 56
```
이전처럼 탭 바 기본 높이 56에 노치 유무에 따라 노치 높이 만큼 y값을 올리는 방식으로 구현했습니다.

Minny27 commented on Projects/Core/Src/BaseViewController/TFLaunchVIewController.swift:
```diff
@@ -14,15 +14,17 @@ import DSKit
+  
   public override func loadView() {
     super.loadView()
+    view.backgroundColor = DSKitAsset.Color.neutral700.color
```
스플래시 뷰의 슈퍼 뷰의 백그라운드 칼라를 설정했습니다.

ibcylon commented on Projects/Features/Like/Src/Subview/Cell/LikeProfileCollectionViewCell.swift:
```diff
@@ -48,7 +48,7 @@ final class ProfileCollectionViewCell: UICollectionViewCell {
       return
     }
-    let random: [DSKitImages] = [DSKitAsset.Image.test1, DSKitAsset.Image.test2]
+    let random: [DSKitImages] = [DSKitAsset.Image.Test.test1, DSKitAsset.Image.Test.test2]
```
Test name space 있으니까 좋네요

ibcylon commented on Projects/Features/Like/Src/Home/LikeHomeViewController.swift:
```diff
@@ -40,14 +40,22 @@ public final class LikeHomeViewController: TFBaseViewController {
   }
+  
+  public override func makeUI() {
+    tabBarItem = UITabBarItem(
```
이 부분은 VC에 직접 넣는 것 보다
각 tabbar 피처들의 네비게이션에 설정해주면 됩니다.

코드 위치를 VC보다 tabbar vc로 옮기는 걸 추천합니다. 거기서 일괄적으로 관리할 수 있고, vc는 코드양도 줄고 책임도 덜게 됩니다.

Minny27 committed:
Refactor: 탭 바 아이템 코드 수정

- 탭 바 아이템 생성 코드 tab bar controller에 추가
- vc에 있는 tab bar 설정 코드 삭제
- main coordinator에서 탭 바 설정하는 코드 추가
- 들여쓰기 설정

Minny27 commented on Projects/Features/Src/Main/TFTabBarViewController.swift:
```diff
@@ -61,3 +61,59 @@ final class TFTabBarController: UITabBarController, MainViewControllable {
+    }
+  }
+  
+  static func makeTabItem(_ item: TabItem) -> UITabBarItem {
```
tab bar controller에 확장으로 탭 바 아이템 생성하는 코드 추가했습니다.

Minny27 commented on Projects/Features/Src/Main/Coordinator/MainCoordinator.swift:
```diff
@@ -27,26 +27,26 @@ final class MainCoordinator: BaseCoordinator, MainCoordinating {
     let likeCoordinator = likeBuildable.build(rootViewControllable: NavigationViewControllable())
     attachChild(likeCoordinator)
-    likeCoordinator.start()
+    likeCoordinator.viewControllable.uiController.tabBarItem = .makeTabItem(.like)
```
제안해주신 방법대로 MainCoordinator에  탭 바 설정하는 코드추가했습니다.

ibcylon reviewed (approved):
수고하셨습니다~
